packages/accounts/src/light-account/account.ts

```diff
@@ -160,74 +192,148 @@ export async function createLightAccount({
     });
   };
 
-  const account = await toSmartContractAccount({
-    transport,
-    chain,
-    entryPoint,
-    accountAddress: address,
-    source: "LightAccount",
-    getAccountInitCode,
-    encodeExecute: async ({ target, data, value }) => {
-      return encodeFunctionData({
-        abi: LightAccountAbi,
-        functionName: "execute",
-        args: [target, value ?? 0n, data],
-      });
-    },
-    encodeBatchExecute: async (txs) => {
-      const [targets, values, datas] = txs.reduce(
-        (accum, curr) => {
-          accum[0].push(curr.target);
-          accum[1].push(curr.value ?? 0n);
-          accum[2].push(curr.data);
+  const account =
```

Comment:
there's a lot of duplicated code here. We should leave `createLightAccount` as it was and have it just require 0.6 for now. in a later PR we should tackle supporting LightAccount v2 and do it in such a way that we're not duplicating a lot of code

Reply:
I agree about the duplicated code, but the contracts team is also hoping that they can use the SDK to test light account v2 so they would like support here even before it's deployed. I'll get some more info about what they're looking for.

Reply:
makes sense, we should do the 0.7 support for LA as a separate PR to this so that it's easier to parse and review